Src/AutoFixture/InjectOnMatchCustomization.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using AutoFixture.Kernel;
+
+namespace AutoFixture
+{
+    /// <summary>
+    /// A customization that injects a specific value of a specific <see cref="Type"/>
+    /// and uses them to satisfy requests that match a set of criteria.
+    /// </summary>
+    /// <remarks>
+    /// The criteria that determine whether the frozen specimen will be used
+    /// to satisfy a given request are specified at construction time
+    /// as a <see cref="IRequestSpecification"/> object.
+    /// Multiple criteria can be combined together as a boolean expression
+    /// by composing them into a <see cref="AndRequestSpecification"/>
+    /// or <see cref="OrRequestSpecification"/> object.
+    /// </remarks>
+    public class InjectOnMatchCustomization : ICustomization
```

Comment:
To me this name is misleading a bit. From the name I would expect that something would be injected somewhere when a matching request is coming. However, value is _already_ injected by a fact of adding this customization. From this perspective I would say `ReturnOnMatchCustomization` or `FixedValueOnMatchCustomization` is more clear for the functionality this class provides.

Another aspect is that customization itself is not a universal building block and I don't see how it can be used by clients. If they reference `AutoFixture` directly, then they would just use `fixture.Inject<T>()` method. Personally I think we should add only truly reusable building blocks to the core. Otherwise, it might become just a dump of junk customizations that are needed once in a while in very exotic situations 😟

I would just inline this one-liner in the `InjectedAttribute` directly:

```c#
fixture.Customizations.Insert(0, FilteringSpecimenBuilder(new FixedBuilder(value), this.Matcher));
```

Reply:
I agree very much with you. I just mirrored how the `Frozen` attribute was implemented :)

Anyway, as per @aivascu comment, we're changing approach. So these changes will be removed in the next commit :)